Lib/concurrent/futures/thread.py

```diff
@@ -129,6 +136,8 @@ def __init__(self, max_workers=None, thread_name_prefix='',
 
         self._max_workers = max_workers
         self._work_queue = queue.SimpleQueue()
+        self._idle_lock = threading.Lock()
+        self._idle_count = 0
```

Comment:
Why don't you use a `threading.Semaphore` for  this?
It makes the implementation easier to read as you don't need to handle the lock + increments but only rely on `acquire` and `release`.